tom/main.py

```diff
@@ -2,53 +2,101 @@
 import argparse
 import logging as log
 
-from tom.bot import Tom
-
-
-def run_tom(interactive, secrets_dir, talk_mode):
-    secrets = {}
-    names = [
-        "GITHUB_TOKEN", "JENKINS_CRUMB", "JENKINS_USER", "JENKINS_TOKEN", "SLACK_READ_TOKEN",
-        "SLACK_SEND_TOKEN", "SLACK_APP_TOKEN"
-    ]
-    for n in names:
-        secrets[n] = get_var(n, secrets_dir)
-    tom = Tom(secrets, interactive)
-    if talk_mode:
-        tom.talk()
-    else:
-        tom.run()
+from tom.bot import Bot
+from tom.utils import read_json, user_error
 
 
-def get_var(name, dir="./"):
-    var = None
-    if name in os.environ:
-        var = os.environ[name]
-    else:
-        try:
-            with open(os.path.join(dir, name), "r") as f:
-                var = f.read().strip()
-        except (PermissionError, FileNotFoundError):
-            pass
-    return var
+def setup_bot(directory, interactive, data):
+    username = data["username"]
+    secrets = data["secrets_data"]
+
+    orgs = data["orgs"]
+    repos = data["repos"]
+    jenkins = data["jenkins"]
+    job = data["jenkins_job"]
+    reviewers = data["reviewers"]
+    trusted = data["trusted"]
+
+    return Bot(
+        username, secrets, orgs, repos, jenkins, job, reviewers, trusted, directory, interactive)
+
+
+def run_talk(directory, user):
+    config = load_config(directory)
+    assert len(config["bots"]) > 0
+    for bot_data in config["bots"]:
+        if bot_data["username"] == user:
+            bot = setup_bot(directory, False, bot_data)
+            bot.talk()
+            return
+    user_error("Couldn't find config for bot '{}'".format(user))
+
+
+def run_bot(directory, interactive, data):
+    bot = setup_bot(directory, interactive, data)
+    bot.run()
+
+
+def load_config(directory):
+    config = read_json(os.path.join(directory, "config.json"))
+    assert config
+
+    for bot_data in config["bots"]:
+
+        secrets_path = bot_data["secrets"]
+        if not secrets_path.startswith("/"):
+            secrets_path = os.path.join(directory, secrets_path)
```

Comment:
`os.path.join()` does this weird thing on its own -- it ignores the first arg if the second one starts with `/`

Reply:
Yeah, I'm all for that.